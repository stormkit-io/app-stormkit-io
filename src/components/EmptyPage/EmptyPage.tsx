import type { SxProps } from "@mui/material";
import React from "react";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import emptyListSvg from "~/assets/images/empty-list.svg";

interface Props {
  sx?: SxProps;
  children?: React.ReactNode;
}

export default function EmptyList({ sx, children }: Props) {
  return (
    <Box sx={{ textAlign: "center", my: 12, ...sx }}>
      <img src={emptyListSvg} alt="Empty app list" className="m-auto" />
      <Typography fontSize="medium" sx={{ mt: 6 }}>
        {children || <>It's quite empty in here.</>}
      </Typography>
    </Box>
  );
}
