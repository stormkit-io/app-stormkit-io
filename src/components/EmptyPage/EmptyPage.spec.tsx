import type { SxProps } from "@mui/material";
import { describe, expect, it } from "vitest";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import EmptyPage from "./EmptyPage";

interface Props {
  children: React.ReactNode;
  sx: SxProps;
}

describe("~/components/EmptyPage.tsx", () => {
  let wrapper: RenderResult;

  const createWrapper = ({ children, sx }: Props) => {
    wrapper = render(<EmptyPage children={children} sx={sx} />);
  };

  it("should render the component as expected", () => {
    createWrapper({
      children: <span>Hello world</span>,
      sx: { color: "black" },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
