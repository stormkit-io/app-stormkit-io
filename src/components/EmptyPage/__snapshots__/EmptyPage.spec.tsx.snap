// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`~/components/EmptyPage.tsx > should render the component as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="side-bar-root"
    />
    <div>
      <div
        class="MuiBox-root css-1lo9ce4"
      >
        <img
          alt="Empty app list"
          class="m-auto"
          src="/src/assets/images/empty-list.svg"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-ugwmb-MuiTypography-root"
        >
          <span>
            Hello world
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1lo9ce4"
    >
      <img
        alt="Empty app list"
        class="m-auto"
        src="/src/assets/images/empty-list.svg"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-ugwmb-MuiTypography-root"
      >
        <span>
          Hello world
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
